File: 2. Loop Problems/Loop problems in python/p14.py
# Python program to calculate the factorial of a number


number = int(input("Enter a number: "))


if number < 0:
    print("Factorial of a negative number is not defined.")
else:
    factorial = 1  
   
    for i in range(1, number + 1):
        factorial *= i
    print(f"Factorial of {number} = {factorial}")

File: 2. Loop Problems/Loop Problems in c/p14.c
// Write a C program to calculate factorial of a number

#include <stdio.h>

int main() {
    int number;
    long long factorial = 1; 

   
    printf("Enter a number: ");
    scanf("%d", &number);

    
    if (number < 0) {
        printf("Invalid Input as negative numbers don't have Factorial.\n");
    } 
    
    else {
        
        for (int i = 1; i <= number; i++) {
            factorial *= i;
        }
        printf("Factorial of %d = %lld\n", number, factorial);
    }

    
}

